component/azstorage/config.go

```diff
@@ -450,14 +453,15 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		az.stConfig.authConfig.ResourceID = opt.ResourceID
 	case EAuthType.SPN():
 		az.stConfig.authConfig.AuthMode = EAuthType.SPN()
-		if opt.ClientID == "" || (opt.ClientSecret == "" && opt.OAuthTokenFilePath == "") || opt.TenantID == "" {
+		if opt.ClientID == "" || (opt.ClientSecret == "" && opt.OAuthTokenFilePath == "" && opt.WorkloadIdentityToken == "") || opt.TenantID == "" {
 			//lint:ignore ST1005 ignore
-			return errors.New("Client ID, Tenant ID or Client Secret not provided")
+			return errors.New(fmt.Sprintf("Client ID %s, Tenant ID %s or Client Secret %s, OAuthTokenFilePath %s, WorkloadIdentityToken %s not provided", opt.ClientID, opt.TenantID, opt.ClientSecret, opt.OAuthTokenFilePath, opt.WorkloadIdentityToken))
```

Comment:
We should not log the secret or token here

Reply:
Ok, fixed